src/supa/job/shared.py

```diff
@@ -155,3 +158,11 @@ def text(self) -> str:
     def __str__(self) -> str:
         """Return the exception in a human readable format."""
         return self.text
+
+
+def send_service_exception(nsi_exc: NsiException) -> None:
```

Comment:
Hmm not sure if this implementation is what I meant. This is a function that basically does the same as Raising an exception. 

There usually also is a way to register a function in the app that handles the raising of a certain type of exception. Example, this function would cast a Value error to an HTTPException


### core
```python
def value_error_handler(exc: ValueError, *args, **kwargs) -> HTTPException:
    return HTTPException(status_code=HTTPStatus.BAD_REQUEST, detail=str(exc))


app.register_exception_handler(ValueError, value_error_handler)
```


### Somewhere else in the app

```python
...
   raise ValueError("My value error")
....
```

This error will be handled by the exception_handler.

Perhaps this already exists for the `NSIException` ?

Reply:
The `send_service_exception()` performs a gRPC ServiceException call to PolyNSI that will convert it to a SOAP `ServiceException`. Packaging the information in a `NsiException` feels appropriate here, and may come in handy later when other raised `NsiException`s are caught and send as `ServiceException` to PolyNSI as well. 

Added the implementation of `send_service_exception()`.